### ISSUE TITLE
5.4.0 Release introduced GCC warning

### ISSUE THREAD

**Issue by kohtenko** [hamcrest/OCHamcrest#79]
This warning appeared after 5.4.0 release
```
HCIsEqualCompressingWhiteSpace.h:43:85: GCC does not allow 'deprecated' attribute in this position on a function definition
```

**Comment by jonreid** [comment]
I should have put this in the release notes as a known issue. The problem will go away of course when I remove the deprecated functions in 6.0.0.

If you know a way to deprecate a static inline, please let me know. It'll be important going forward.

**Comment by jonreid** [comment]
Note to future self: -Wgcc-compat

**Comment by jonreid** [comment]
Fixed in c79d34c
